test/javascripts/turbolinks_test.coffee

```diff
@@ -122,243 +122,175 @@ describe 'Turbolinks', ->
+        visit url: 'singleScriptInHeadWithDifferentSourceButSameName', ->
+          assert(Turbolinks.fullPageNavigate.called, 'Should perform a full page refresh.')
+          done()
 
-      visit url: 'noScriptsOrLinkInHead', ->
-        assertLinks([])
-        visit url: 'singleLinkInHead', ->
-          assertLinks(['foo.css'])
-          assert.equal(linkTagInserted.callCount, 1)
+    it 'does not refresh page when new data-turbolinks-track values encountered', (done) ->
+      visit url: 'singleScriptInHead', ->
+        visit url: 'twoScriptsInHead', ->
+          assert(Turbolinks.fullPageNavigate.notCalled, 'Should not perform a full page refresh.')
           done()
 
-    it 'dispatches page:after-link-removed event when removing a link on navigation', (done) ->
-      linkTagRemoved = sinon.spy()
-      $(document).on 'page:after-link-removed', linkTagRemoved
+    describe 'using data-turbolinks-track="true"', ->
+      startFromFixture = (route) ->
+        fixtureHTML = ROUTES[route][2]
+        $fixture = $(fixtureHTML)
+        document.body.innerHTML = $fixture.find('body').html()
+        document.head.innerHTML = $fixture.find('head').html()
 
-      visit url: 'singleLinkInHead', ->
-        visit url: 'noScriptsOrLinkInHead', ->
-          assert.equal(linkTagRemoved.callCount, 1)
+      it 'refreshes page when a new tracked node is present', (done) ->
+        visit url: 'singleScriptInHeadTrackTrue', ->
+          assert(Turbolinks.fullPageNavigate.called, 'Should perform a full page refresh.')
           done()
 
-    it 'inserts link with a new href into empty head on navigation', (done) ->
-      visit url: 'noScriptsOrLinkInHead', ->
-        assertLinks([])
-        visit url: 'singleLinkInHead', ->
-          assertLinks(['foo.css'])
+      it 'refreshes page when an extant tracked node is missing', (done) ->
+        startFromFixture('twoScriptsInHeadTrackTrue')
+        visit url: 'singleScriptInHeadTrackTrue', ->
+          assert(Turbolinks.fullPageNavigate.callCount, 2, 'Should perform two full page refreshes.')
           done()
 
-    it 'inserts link with a new href into existing head on navigation', (done) ->
-      visit url: 'singleLinkInHead', ->
-        assertLinks(['foo.css'])
-        visit url: 'twoLinksInHead', ->
-          assertLinks(['foo.css', 'bar.css'])
+      it 'does not refresh page when tracked node are the same', (done) ->
+        startFromFixture('singleScriptInHeadTrackTrue')
+        visit url: 'singleScriptInHeadTrackTrue', ->
+          assert.equal(Turbolinks.fullPageNavigate.callCount, 1, 'Should not perform a full page refresh.')
           done()
 
-    it 'inserts link with a new href before exiting scripts in head on navigation', (done) ->
-      visit url: 'singleLinkInHead', ->
-        assertLinks(['foo.css'])
-        visit url: 'twoLinksInHeadReverseOrder', ->
-          assertLinks(['bar.css', 'foo.css'])
-          done()
+    describe 'link tags', ->
```

Comment:
Because it fits the conventions of the rest of the lib. We fire `after-node-removed` events and such already. I think going forward we can probably afford to have a conversation about the event lifecycle and what's really needed, I could take or leave these ones for now though.